video2dataset/workers/subset_worker.py

```diff
@@ -67,54 +118,102 @@ def process_shard(
         failed_to_subsample = 0
         error_message = None
 
+        if "s3" in shard:
+            shard = f"pipe:aws s3 cp {shard} -"
+
         dataloader = get_video_dataset(
             urls=[shard],
             batch_size=1,
             decoder_kwargs={},
+            enforce_additional_keys=[],
         )
+        count = 0
         for sample in dataloader:
             # Gather subset of dataset
             key = sample["__key__"]
+            count += 1
             caption = sample.get("txt", b"").decode("utf-8")
             meta = json.loads(sample.get("json", b"{}").decode("utf-8"))
-
             streams = {}
             for mod, fmt in self.encode_formats.items():
                 streams[mod] = sample[fmt]
 
+            if self.captions_are_subtitles:  # create clips
+                subtitles = meta["yt_meta_dict"]["subtitles"]
+                meta["clips"] = [[line_dict["start"], line_dict["end"]] for line_dict in subtitles]
+                meta["lines"] = [" ".join(line_dict["lines"]) for line_dict in subtitles]
+
+            elif self.detect_cuts:  # apply cut detection to get clips
+                detected_cuts = self.cut_detector(streams)
+                meta["cuts"] = detected_cuts
```

Comment:
can we try to make the formatting between this file and download_subsampler as similar as possible? f.e. here you just do ```meta["cuts"] = self.cut_detector(streams)```

it's easier to review and will be easier later when we unify all workers

Reply:
there are other cases in this file like in __init__, small things like ordering of when you do self.something = something but its much easier when you come back to it after some time